shell/server/pre-meteor.js

```diff
@@ -210,7 +210,24 @@ function checkMagic(buf, magic) {
   return true;
 }
 
-function serveStaticAsset(req, res) {
+function serveSelfTest(req, res) {
+  inMeteor(() => {
+    if (req.method === 'GET' &&
+        req.url === '/') {
+      return serveStaticAsset(req, res, {
```

Comment:
I agree this should not use the `static` subdomain, since all kinds of special-purpose behavior might apply to it in the future. I'd prefer using an independent subdomain like `self-test`. You don't need to handle `OPTIONS` -- for a `GET` request, setting `Access-Control-Allow-Origin: *` in the response is all you need.